pkg/apis/componentconfig/v1alpha1/temp_v1a1_refactor.gogo

```diff
@@ -0,0 +1,548 @@
+// to dynamic config. We can always eleveate these later if we need to.
+type ShouldProbablyJustBeFlags struct {
+  /* Static pod stuff should all just be flags */
+  // podManifestPath is the path to the directory containing pod manifests to
+  // run, or the path to a single manifest file
+  PodManifestPath string `json:"podManifestPath"`
+  // fileCheckFrequency is the duration between checking config files for
+  // new data
+  FileCheckFrequency metav1.Duration `json:"fileCheckFrequency"`
+  // httpCheckFrequency is the duration between checking http for new data
+  HTTPCheckFrequency metav1.Duration `json:"httpCheckFrequency"`
+  // manifestURL is the URL for accessing the container manifest
+  ManifestURL string `json:"manifestURL"`
+  // manifestURLHeader is the HTTP header to use when accessing the manifest
+  // URL, with the key separated from the value with a ':', as in 'key:value'
+  ManifestURLHeader string `json:"manifestURLHeader"`
+
+  // maxOpenFiles is Number of files that can be opened by Kubelet process.
+  MaxOpenFiles int64 `json:"maxOpenFiles"`
+
+  // experimentalMounterPath is the path to mounter binary. If not set, kubelet will attempt to use mount
+  // binary that is available via $PATH,
+  ExperimentalMounterPath string `json:"experimentalMounterPath,omitempty"`
+
+  // containerized should be set to true if kubelet is running in a container.
+  Containerized *bool `json:"containerized"`
+  // lockFilePath is the path that kubelet will use to as a lock file.
+  // It uses this file as a lock to synchronize with other kubelet processes
+  // that may be running.
+  LockFilePath *string `json:"lockFilePath"`
+  // ExitOnLockContention is a flag that signifies to the kubelet that it is running
+  // in "bootstrap" mode. This requires that 'LockFilePath' has been set.
+  // This will cause the kubelet to listen to inotify events on the lock file,
+  // releasing it and exiting when another process tries to open that file.
+  ExitOnLockContention bool `json:"exitOnLockContention"`
+
+  // rootDirectory is the directory path to place kubelet files (volume
+  // mounts,etc).
+  RootDirectory string `json:"rootDirectory"`
+
+  // containerRuntime is the container runtime to use.
+  ContainerRuntime string `json:"containerRuntime"`
+  // remoteRuntimeEndpoint is the endpoint of remote runtime service
+  RemoteRuntimeEndpoint string `json:"remoteRuntimeEndpoint"`
+  // remoteImageEndpoint is the endpoint of remote image service
+  RemoteImageEndpoint string `json:"remoteImageEndpoint"`
+  // runtimeRequestTimeout is the timeout for all runtime requests except long running
+  // requests - pull, logs, exec and attach.
+  RuntimeRequestTimeout metav1.Duration `json:"runtimeRequestTimeout"`
+}
+
+/* BEGIN RUNTIME REMOVAL */
+
+// I believe the plan is to remove ALL of these from the Kubelet/Node configuration
+// e.g. they become flags-only for now and are immediately marked deprecated, for later removal.
```

Comment:
I don't think we can deprecate these flags anytime soon because there are no alternatives to specify these settings. We can only deprecate them when/if we deprecate the docker integration and/or enable running dockershim as a separate process.